### ISSUE TITLE
Deploy custom image with the kube-deploy make target

### ISSUE THREAD

**Issue by arkodg** [envoyproxy/gateway#323]
*Description*:
`make kube-deploy` deploys the Gateway API CRDs as well as the EG resources into the K8s Cluster
It however deploys the `envoyproxy/gateway-dev:latest` image 
https://github.com/envoyproxy/gateway/blob/6fd629a98a5149e99f69e68337d1d8404ce35ebf/internal/provider/kubernetes/config/envoy-gateway/deploy_and_ns.yaml#L31

Allowing the user to pass an ARG/ENV to `make kube-deploy` such as `IMAGE="foo/bar:latest" make kube-deploy` would allow devs to deploy their custom images in the cluster and test it out

**Comment by danehans** [comment]
The kube-install-image Make target builds a local image and pushes it to the cluster.

**Comment by arkodg** [comment]
> The kube-install-image Make target builds a local image and pushes it to the cluster.

this command builds & pushes into the kind cluster, this is very helpful for dev workflows
but also another case is to push to a non kind cluster.

TLDR we need to find a better way to `kustomize` image names & tag / maybe we can reuse https://github.com/envoyproxy/gateway/blob/c0a21c051488d90c291a4b75f7c5de07b62bd49a/tools/hack/release-manifests.sh#L34

**Comment by Xunzhuo** [comment]
@arkodg /assign, I can work on this:)
